modules/unsupported/vector-mosaic/src/main/java/org/geotools/vectormosaic/VectorMosaicStore.java

```diff
@@ -0,0 +1,484 @@
+            while (iterator.hasNext()) {
+                SimpleFeature delegateFeature = iterator.next();
+                String typeName =
+                        delegateFeature
+                                .getAttribute(VectorMosaicGranule.TYPE_NAME_DELEGATE_FIELD)
+                                .toString();
+                if (typeMap.containsKey(typeName)) {
+                    VectorMosaicTypeConfiguration typeConfiguration = typeMap.get(typeName);
+                    typeConfiguration.addGranule(
+                            VectorMosaicGranule.fromDelegateFeature(delegateFeature));
+                } else {
+                    VectorMosaicTypeConfiguration typeConfiguration =
+                            new VectorMosaicTypeConfiguration(
+                                    typeName,
+                                    VectorMosaicGranule.fromDelegateFeature(delegateFeature));
+                    typeMap.put(typeName, typeConfiguration);
+                }
+            }
+        } catch (IOException e) {
+            LOGGER.log(
+                    Level.WARNING, "Failed to get feature source for type " + delegateTypeName, e);
+        } finally {
+            if (iterator != null) {
+                iterator.close();
+            }
+        }
+    }
+
+    /**
+     * Validates the schema of the delegate store.
+     *
+     * @param delegateStoreName the name of the delegate store.
+     * @param typeName the name of the type to validate.
+     * @param delegateSchema the schema of the type to validate.
+     * @return true if the schema is valid.
+     */
+    private boolean validateSchema(
+            String delegateStoreName, String typeName, SimpleFeatureType delegateSchema) {
+        for (String field : REQUIRED_VECTOR_MOSAIC_FIELDS.keySet()) {
+            if (delegateSchema.getDescriptor(field) == null) {
+                LOGGER.log(
+                        Level.INFO,
+                        "Delegate store "
+                                + delegateStoreName
+                                + " schema with type name "
+                                + typeName
+                                + " does not contain required field "
+                                + field);
+                return false;
+            } else if (!delegateSchema
+                    .getDescriptor(field)
+                    .getType()
+                    .getBinding()
+                    .getTypeName()
+                    .equals(REQUIRED_VECTOR_MOSAIC_FIELDS.get(field))) {
+                LOGGER.log(
+                        Level.INFO,
+                        "Delegate store "
+                                + delegateStoreName
+                                + " schema with type name "
+                                + typeName
+                                + " has the wrong field type for field "
+                                + field);
+                return false;
```

Comment:
Iterate on the map entries instead, put the descriptor in a local variable, to reduce the verboseness of this method.

Reply:
Fixed